shared/src/commonMain/kotlin/org/hyperskill/app/analytic/domain/processor/AmplitudeAnalyticEventMapper.kt

```diff
@@ -0,0 +1,40 @@
+package org.hyperskill.app.analytic.domain.processor
+
+import org.hyperskill.app.analytic.domain.model.AnalyticEvent
+import org.hyperskill.app.analytic.domain.model.amplitude.AmplitudeAnalyticEvent
+import org.hyperskill.app.analytic.domain.model.hyperskill.HyperskillAnalyticEvent
+import org.hyperskill.app.analytic.domain.model.hyperskill.HyperskillAnalyticKeys
+
+internal object AmplitudeAnalyticEventMapper {
+
+    fun map(analyticEvent: AnalyticEvent): AmplitudeAnalyticEvent? =
+        when (analyticEvent) {
+            is HyperskillAnalyticEvent -> AmplitudeAnalyticEvent(
+                name = getType(analyticEvent),
+                params = getParams(analyticEvent)
+            )
+            is AmplitudeAnalyticEvent -> analyticEvent
+            else -> null
+        }
+
+    private fun getType(analyticEvent: HyperskillAnalyticEvent): String =
```

Comment:
I suppose result string should be with no spaces or new lines in beginning and end

Reply:
@ivan-magda 
As for task description, "action, part & target should be concatenated using space char".
But, yeah, in some cases there will be a redundant space.

Reply:
I suppose we should build string smth like this:
```kotlin
buildString {
    append(analyticEvent.action.actionName)
    if (analyticEvent.part != null) {
        append(' ')
        append(analyticEvent.part.partName)
    }
    if (analyticEvent.target != null) {
        append(' ')
        append(analyticEvent.target.targetName)
    }
}
```

Reply:
@ivan-magda 
Yeah, your are right